### ISSUE TITLE
👀 Tracking: Improve accessibility (a11y) for the website

### ISSUE THREAD

**Issue by RogueArt** [uclaacm/linux-lab#123]
### What is the issue?

Currently, there is little focus on accessibility throughout the project.
Breaking this down further, there is a lack of:
- Accessibility labels (e.g. input labels, form labels, etc.)
- Semantic HTML for users using screen readers
- Keyboard accessibility (shortcuts, non-mouse ways of interacting)

### Why should we spend developer time on this?
While the vast majority of users of Linux Lab may not need this, improving accessibility will improve the experience across all users, and it will greatly improve the experience for users that need it the most. Also, it's the right thing to do.

### What are some examples of this?
- [Modal.tsx](https://github.com/uclaacm/linux-lab/blob/main/src/components/shared/Modal.tsx) uses almost completely `div` elements, we could improve semantic HTML here by using `article`, `section`, or even `table` elements
- [ProgressBar.tsx](https://github.com/uclaacm/linux-lab/blob/main/src/components/shared/progressbar.tsx) - If I am using a screen reader, how do I know what my progress is? There is no label or element that tells me, this is purely visual

### Checklist:
- [ ] Add accessibility labels to current code, keeping in mind users have screen readers
- [ ] Refactor `div` and other generic elements to follow semantic HTML
- [ ] Add keyboard shortcuts to navigate the page
- [ ] Verify we can tab across and to elements
- [ ] Set up some form of accessibility testing / validation to make sure we are meeting all requirements

This is a 👀 Tracking Issue
